vendor/github.com/gorilla/websocket/client.go

```diff
@@ -303,6 +303,12 @@ func (d *Dialer) DialContext(ctx context.Context, urlStr string, requestHeader h
 			return nil, nil, err
 		}
 		if proxyURL != nil {
+			if proxyURL.Scheme == "https" {
+				netDial = func(unused, addr string) (net.Conn, error) {
```

Comment:
I _think_ this needs to use the same configured dialers if provided used to calculate https dial above:

```
var proxyDialer func(network, add string) (net.Conn, error)
if proxyURL.Scheme == "https" {
	if d.NetDialTLSContext != nil {
		netDial = func(network, addr string) (net.Conn, error) {
			return d.NetDialTLSContext(ctx, network, addr)
		}
	} else if d.NetDialContext != nil {
		netDial = func(network, addr string) (net.Conn, error) {
			... use d.NetDialContext to dial and verify TLS using TLSClientConfig
		}
	} else if d.NetDial != nil {
		netDial = ... use d.NetDial to dial and verify TLS using TLSClientConfig
	}
} else {
  proxyDialer = netDial
}
```